pkg/dcl/clientconfig/config.go

```diff
@@ -108,11 +122,11 @@ func NewForIntegrationTest() *dcl.Config {
 		opt.HTTPClient = &http.Client{Transport: t}
 	}
 
-	config, err := New(ctx, opt)
+	config, httpClient, err := NewWithHttpClient(ctx, opt)
```

Comment:
It looks like you don't really need `NewWithHttpClient`, because the only place it's called where we look for the http client return value is in this method, and opt.HTTPClient is used instead.  Maybe just don't export `NewWithHTTPClient` (i.e. rename to `newWithHTTPClient`?)